src/consensus.rs

```diff
@@ -70,17 +70,29 @@ pub(crate) async fn run<H: Hasher + 'static>(
         .fuse();
     info!(target: "AlephBFT", "{:?} All services started.", conf.node_ix);
 
+    let mut terminal_exited = false;
+    let mut creator_exited = false;
+    let mut extender_exited = false;
     futures::select! {
         _ = exit => {},
-        _ = terminal_handle => {},
-        _ = creator_handle => {},
-        _ = extender_handle => {},
+        _ = terminal_handle => terminal_exited = true,
```

Comment:
I think it makes sense to add debug messages here saying something like "Terminal task terminated early". Might help a lot sometime.

Reply:
Added.